config/metric.go

```diff
@@ -208,12 +208,48 @@ func otlpGRPCMetricExporter(ctx context.Context, otlpConfig *OTLPMetric) (sdkmet
 		opts = append(opts, otlpmetricgrpc.WithTimeout(time.Millisecond*time.Duration(*otlpConfig.Timeout)))
 	}
 	if len(otlpConfig.Headers) > 0 {
-		opts = append(opts, otlpmetricgrpc.WithHeaders(otlpConfig.Headers))
+		opts = append(opts, otlpmetricgrpc.WithHeaders(toStringMap(otlpConfig.Headers)))
 	}
 
 	return otlpmetricgrpc.New(ctx, opts...)
 }
 
+// newIncludeExcludeFilter returns a Filter that includes attributes
+// in the include list and excludes attributes in the excludes list.
+// It returns an error if an attribute is in both lists
+//
+// If IncludeExclude is empty a include-all filter is returned.
+func newIncludeExcludeFilter(lists *IncludeExclude) (attribute.Filter, error) {
+	if lists == nil {
+		return func(kv attribute.KeyValue) bool { return false }, nil
+	}
+
+	included := make(map[attribute.Key]struct{})
+	for _, k := range lists.Included {
+		included[attribute.Key(k)] = struct{}{}
+	}
+	excluded := make(map[attribute.Key]struct{})
+	for _, k := range lists.Excluded {
+		if _, ok := included[attribute.Key(k)]; ok {
+			return nil, fmt.Errorf("attribute cannot be in both include and exclude list: %s", k)
+		}
+		excluded[attribute.Key(k)] = struct{}{}
+	}
```

Comment:
nit: you could use the same map for both of these given you can detect the unset state. e.g.

```go
state := make(map[attribute.Key]bool)
// init ^.
allowed, ok := state[k]
if !ok {
	// unknown
	return true
}
return allowed

Reply:
Right, but an unset included is different from an unset excluded which is how i ended up with two lists.